deeplinks-plugin/src/main/kotlin/com/freeletics/khonshu/deeplinks/plugin/DeepLinkDefinitions.kt

```diff
@@ -0,0 +1,63 @@
+package com.freeletics.khonshu.deeplinks.plugin
+
+import com.freeletics.khonshu.navigation.deeplinks.DeepLinkHandler
+import com.freeletics.khonshu.navigation.deeplinks.replacePlaceholders
+import com.freeletics.khonshu.navigation.internal.InternalNavigationApi
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.decodeFromString
+import net.peanuuutz.tomlkt.Toml
+
+@Serializable
+public data class DeepLinkDefinitions(
+    val prefixes: List<PrefixDefinition> = emptyList(),
+    val placeholders: List<PlaceholderDefinition> = emptyList(),
+    val deepLinks: Map<String, DeepLinkDefinition>,
+) {
+    public companion object {
+        public fun decodeFromString(string: String): DeepLinkDefinitions {
+            return Toml.decodeFromString<DeepLinkDefinitions>(string)
+        }
+    }
+}
+
+@Serializable
+public data class DeepLinkDefinition(
+    val patterns: List<PatternDefinition>,
+    val prefixes: List<PrefixDefinition> = emptyList(), // use global if empty
+    val placeholders: List<PlaceholderDefinition> = emptyList(), // use global if empty or if key not found
+    val exampleQueries: List<String> = emptyList(),
```

Comment:
This is only here for our list/test script. Currently we have deep links with query parameters defined as separate deep links which doesn't work well. This makes it possible to define query parameters here that can be used in examples.